addOns/alertFilters/src/main/java/org/zaproxy/zap/extension/alertFilters/GlobalAlertFilterParam.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2019 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.alertFilters;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.configuration.ConversionException;
+import org.apache.commons.configuration.HierarchicalConfiguration;
+import org.apache.log4j.Logger;
+import org.zaproxy.zap.common.VersionedAbstractParam;
+
+public class GlobalAlertFilterParam extends VersionedAbstractParam {
+
+    private static final Logger logger = Logger.getLogger(GlobalAlertFilterParam.class);
+
+    /**
+     * The version of the configurations. Used to keep track of configurations changes between
+     * releases, if updates are needed.
+     *
+     * <p>It only needs to be updated for configurations changes (not releases of the add-on).
+     */
+    private static final int PARAM_CURRENT_VERSION = 1;
+
+    private static final String GLOBAL_ALERT_FILTERS_BASE_KEY = "globalalertfilter";
+
+    private static final String ALL_ALERT_FILTERS_KEY =
+            GLOBAL_ALERT_FILTERS_BASE_KEY + ".filters.filter";
+
+    private static final String FILTER_RULE_ID_KEY = "ruleid";
+    private static final String FILTER_URL_KEY = "url";
+    private static final String FILTER_URL_IS_REGEX_KEY = "urlregex";
+    private static final String FILTER_NEW_RISK_KEY = "newrisk";
+    private static final String FILTER_PARAMETER_KEY = "param";
+    private static final String FILTER_PARAMETER_IS_REGEX_KEY = "paramregex";
+    private static final String FILTER_ATTACK_KEY = "attack";
+    private static final String FILTER_ATTACK_IS_REGEX_KEY = "attackregex";
+    private static final String FILTER_EVIDENCE_KEY = "evidence";
+    private static final String FILTER_EVIDENCE_IS_REGEX_KEY = "evidenceregex";
+    private static final String FILTER_ENABLED_KEY = "enabled";
+
+    private static final String CONFIRM_REMOVE_FILTER_KEY =
+            GLOBAL_ALERT_FILTERS_BASE_KEY + ".confirmRemoveFilter";
+
+    private List<AlertFilter> alertFilters = null;
```

Comment:
Since we don't allow/want duplicates it could use a `Set` (also avoids the need to check if they are already contained).